src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,23 +34,10 @@ public static boolean isValidAddress(String test) {
     }
```

Comment:
The constructor and the validation method are shared by all the classes, so better to abstract them as well.